INT-89: Check user login state before allowing edits

perjg opened the pull request:
Checks the user's login state before allowing edits.

Logged in users are always allowed.
Anonymous edits are allowed unless disableAnonymousEditing is set to true
Any wikia marked as a COPPA wikia always requires login to edit regardless of settings.

Depends on: https://github.com/Wikia/app/pull/8896

hakubo commented on front/scripts/main/components/ArticleWrapperComponent.ts:
```diff
@@ -86,6 +86,27 @@ App.ArticleWrapperComponent = Em.Component.extend(
+
+			if (isLoggedIn) {
+				return true;
+			}
```
I think we use: 
} else {

rogatty commented on front/scripts/main/components/ArticleWrapperComponent.ts:
```diff
@@ -86,6 +86,26 @@ App.ArticleWrapperComponent = Em.Component.extend(
+		 * - Wikias with disableAnonymousEditing set need login to edit
+		 * - Coppa wikias (for wikias directed at children) always require login
+		 *
+		 * @returns True if edit is allowed
```
`@returns {boolean} True if edit is allowed`

rogatty commented on front/templates/main/components/article-contribution.hbs:
```diff
@@ -1,7 +1,13 @@
+			{{svg 'pencil' role='img' class='icon pencil'}}
+		</span>
+	{{else}}
+		<span {{action 'select'}}>
```
`select` action tracks `label: 'add-photo-no-auth'`. I don't think we want to do that when user clicks on the pencil. Also `select` should be renamed to something else as it contains no code related to selecting anything.

perjg commented:
INT-246 builds directly on the changes made here.
I've squashed all the commits here and moved them to a new PR that also includes an extra commit with changes for INT-246:

https://github.com/Wikia/mercury/pull/1605
